backend/functions/views.py

```diff
@@ -1,30 +1,178 @@
+import ast
+import json
+
+import pandas as pd
+from dataset import models as dataset_models
+from projects.models import *
 from rest_framework import status
 from rest_framework.decorators import api_view
 from rest_framework.response import Response
-from dataset import models as dataset_models
-from projects.models import *
 from tasks.models import *
-import json
 
-@api_view(['POST'])
+from .utils import (
+    get_batch_translations_using_indictrans_nmt_api,
+    get_batch_translations_using_google_translate,
+    check_translation_function_inputs,
+)
+
+
+## Utility functions
+def save_translation_pairs(
+    languages,
+    input_sentences,
+    output_dataset_instance_id,
+    batch_size,
+    api_type="indic-trans",
+):
+    """Function to save the translation pairs in the database.
+
+    Args:
+        languages (list): List of output languages for the translations.
+        input_sentences (list(list)): List of input sentences in list format containing the - [input_sentence, input_language, context, quality_status]
+        output_dataset_instance_id (int): ID of the output dataset instance.
+        batch_size (int): Number of sentences to be translated in a single batch.
+        api_type (str): Type of API to be used for translation. (default: indic-trans)
+            Allowed - [indic-trans, google]
+    """
+
+    # Iterate through the languages
+    for output_language in languages:
+
+        # Convert the input_sentences list into a dataframe
+        input_sentences_df = pd.DataFrame(
+            input_sentences,
+            columns=[
+                "sentence_text_id",
+                "corrected_text",
+                "input_language",
+                "context",
+                "quality_status",
+                "metadata",
+            ],
+        )
+
+        # Keep only the clean sentences
+        input_sentences_df = input_sentences_df[
+            (input_sentences_df["quality_status"] == "Clean")
+        ].reset_index(drop=True)
+
+        # Check if the dataframe is empty
+        if input_sentences_df.shape[0] == 0:
+            return {"error": "No clean sentences found. Perform project export first."}
+
+        # Make a sentence list for valid sentences to be translated
+        all_sentences_to_be_translated = input_sentences_df["corrected_text"].tolist()
```

Comment:
Since this block of code is common for all langs, shall we move this outside the loop to avoid running this multiple times?